compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Pipeline.ts

```diff
@@ -198,6 +199,11 @@ function* runWithEnvironment(
   inferTypes(hir);
   yield log({kind: 'hir', name: 'InferTypes', value: hir});
 
+  if (env.config.enableFire) {
+    transformFire(hir);
+    yield log({kind: 'hir', name: 'TransformFire', value: hir});
+  }
+
```

Comment:
we might want to validate hook usage first? if you have a conditional useEffect call this seems like it could insert a conditional useFire() call too, and then you'd get a spurious error in the hooks validation.

at least, let's add a test case for this